otter/convergence/steps.py

```diff
@@ -144,7 +144,7 @@ def report_failure(result):
                 if message is not None:
                     return StepResult.FAILURE, [message]
 
-            return StepResult.RETRY, []
+            return StepResult.RETRY, [result[1]]
```

Comment:
Since this code happens to have extracted `result`, it will be better to use `error` instead of `result[1]` instead.